### PR TITLE
fix bug about gang scheduling

### PR DESCRIPTION
**Pull request by wackxu**
/assign @rongou @terrytangyuan

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/mpi-operator/123)

### CODE REVIEW

File: pkg/controllers/v1alpha2/mpi_job_controller.go [hunk 1]
```diff
@@ -71,16 +71,12 @@ const (
...
-	cpuResourceName     = "cpu"
 	labelGroupName      = "group_name"
 	labelMPIJobName     = "mpi_job_name"
 	labelMPIRoleType    = "mpi_role_type"
 )
 
 const (
-	// SuccessSynced is used as part of the Event 'reason' when an MPIJob is
-	// synced.
-	SuccessSynced = "Synced"
```
Comment (gaocegege) [inline pkg/controllers/v1alpha2/mpi_job_controller.go]: Where do we use the const before? I did not see any deletion about it in the PR.

Comment (wackxu) [reply pkg/controllers/v1alpha2/mpi_job_controller.go]: We use this const to record `SuccessSynced`  event for mpijob and I think `synced successfully` is not worth a event and I delete this line in pr https://github.com/kubeflow/mpi-operator/pull/120

```
	c.recorder.Event(mpiJob, corev1.EventTypeNormal, SuccessSynced, MessageResourceSynced)
```

**Comment by wackxu** [comment]
travis ci failed due to download dep failed

File: pkg/controllers/v1alpha2/mpi_job_controller.go (hunk 1 above)
Comment (gaocegege) [reply pkg/controllers/v1alpha2/mpi_job_controller.go]: LGTM

**Comment by gaocegege** [comment]
I triggered a rebuild.

**Comment by terrytangyuan** [comment]
@wackxu Could you resolve the conflicts?

**Comment by wackxu** [comment]
> @wackxu Could you resolve the conflicts?

Done

File: pkg/controllers/v1alpha2/mpi_job_controller.go [hunk 2]
```diff
@@ -1100,6 +1103,23 @@ func newWorker(mpiJob *kubeflow.MPIJob, desiredReplicas int32) *appsv1.StatefulS
...
+			glog.Warning(errMsg)
+		} else {
+			podSpec.Spec.SchedulerName = gangSchedulerName
+		}
+
+		if podSpec.Annotations == nil {
+			podSpec.Annotations = map[string]string{}
+		}
+		// we create the podGroup with the same name as the mpijob
+		podSpec.Annotations["scheduling.k8s.io/group-name"] = mpiJob.Name
```
Comment (fisherxu) [inline pkg/controllers/v1alpha2/mpi_job_controller.go]: use a const here

Comment (wackxu) [reply pkg/controllers/v1alpha2/mpi_job_controller.go]: @fisherxu Done

Comment (fisherxu) [reply pkg/controllers/v1alpha2/mpi_job_controller.go]: CI seems failed, let's fix it and get merged.

Comment (wackxu) [reply pkg/controllers/v1alpha2/mpi_job_controller.go]: CI is green now, PTAL

**Pull request merged**